astropy/modeling/core.py

```diff
@@ -2741,7 +2741,7 @@ def _check_param_broadcast(self, max_ndim):
 
         # Now check mutual broadcastability of all shapes
         try:
-            check_broadcast(*all_shapes)
+            np.broadcast_shapes(*all_shapes)
         except IncompatibleShapeError as exc:
             shape_a, shape_a_idx, shape_b, shape_b_idx = exc.args
```

Comment:
```suggestion
        except ValueError as exc:
            shape_a, shape_a_idx, shape_b, shape_b_idx = exc.args
```

Since `ValueError` does not have the shapes and indices in the exception args, please parse them from Numpy's error message here:

https://github.com/numpy/numpy/blob/v1.26.1/numpy/core/src/multiarray/iterators.c#L1143-L1146